dotnet/src/Connectors/Connectors.Memory.AzureCosmosDBNoSQL/CosmosSystemTextJSonSerializer.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+// Taken from https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4332
+
+using System;
+using System.IO;
+using System.Reflection;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.Azure.Cosmos;
+
+/// <summary>
+/// This class provides a default implementation of System.Text.Json Cosmos Linq Serializer.
+/// </summary>
+public class CosmosSystemTextJsonSerializer : CosmosLinqSerializer
+{
+    /// <summary>
+    /// A read-only instance of <see cref="JsonSerializerOptions"/>.
+    /// </summary>
+    private readonly JsonSerializerOptions _jsonSerializerOptions;
+
+    /// <summary>
+    /// Creates an instance of <see cref="CosmosSystemTextJsonSerializer"/>
+    /// with the default values for the Cosmos SDK
+    /// </summary>
+    /// <param name="jsonSerializerOptions">An instance of <see cref="JsonSerializerOptions"/> containing the json serialization options.</param>
+    public CosmosSystemTextJsonSerializer(
+        JsonSerializerOptions jsonSerializerOptions)
+    {
+        this._jsonSerializerOptions = jsonSerializerOptions;
+    }
+
+    /// <inheritdoc/>
+    public override T FromStream<T>(Stream stream)
+    {
+        if (stream == null)
+        {
+            throw new ArgumentNullException(nameof(stream));
+        }
+
+        if (stream.CanSeek && stream.Length == 0)
+        {
+            return default;
+        }
+
+        if (typeof(Stream).IsAssignableFrom(typeof(T)))
+        {
+            return (T)(object)stream;
+        }
+
+        using (stream)
+        {
+            using StreamReader reader = new(stream);
+            return JsonSerializer.Deserialize<T>(reader.ReadToEnd(), this._jsonSerializerOptions);
+        }
+    }
+
+    /// <inheritdoc/>
+    public override Stream ToStream<T>(T input)
+    {
+        MemoryStream streamPayload = new();
+        using Utf8JsonWriter writer = new(streamPayload);
+
+        JsonSerializer.Serialize(
```

Comment:
Same here... could this Serialize directly to streamPayload?

Reply:
This overload:
https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.serialize?view=net-8.0#system-text-json-jsonserializer-serialize-1(system-io-stream-0-system-text-json-jsonserializeroptions)
Is that not showing up for some reason?

Reply:
I think you meant 
https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonserializer.serialize?view=net-8.0#system-text-json-jsonserializer-serialize-1(system-io-stream-0-system-text-json-jsonserializeroptions).

It wasn't showing up because of the named parameters.  Fixed now.